R/functions.R

```diff
@@ -315,6 +315,15 @@ alive_identification <- function(input_read_RNA_assay,
     left_join(empty_droplets_tbl, by=".cell") |>
     filter(!empty_droplet)
   
+  # Calculate nFeature_RNA and nCount_RNA if not exist in the data
+  if (!any(str_which(colnames(input_read_RNA_assay[[]]), "nFeature_RNA")) ||
+      !any(str_which(colnames(input_read_RNA_assay[[]]), "nCount_RNA"))) {
+    input_read_RNA_assay[["nFeature_RNA"]] <- apply(input_read_RNA_assay@assays$RNA@counts, 2, function(x) sum(x > 0))
```

Comment:
1. never use @

rather something like Assays(input_read_RNA_assay, "RNA", slot = "counts")

Look at Seurat documentation

I am pretty sure 

(counts > 0) |> colSums()

Is faster